test/integration/controller_instance_test.go

```diff
@@ -992,3 +1055,470 @@ func TestCreateServiceInstanceWithProvisionFailure(t *testing.T) {
+							},
+						},
+						fakeosb.PollLastOperationReaction{
+							Error: errors.New("some error"),
+						},
+						fakeosb.PollLastOperationReaction{
+							Response: &osb.LastOperationResponse{
+								State: osb.StateSucceeded,
+							},
+						},
+					}))
+			},
+			skipVerifyingInstanceSuccess: true,
+			verifyCondition: &v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionTrue,
+				Reason: "ProvisionedSuccessfully",
+			},
+		},
+		{
+			name: "async provisioning with last operation response state failed eventually",
+			setup: func(ct *controllerTest) {
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = fakeosb.DynamicPollLastOperationReaction(getLastOperationResponseByPollCountStates(2, []osb.LastOperationState{osb.StateInProgress, osb.StateFailed}))
+			},
+			skipVerifyingInstanceSuccess: true,
+			verifyCondition: &v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionFalse,
+				Reason: "ProvisionCallFailed",
+			},
+		},
+		{
+			name: "async last operation response successful with originating identity",
+			setup: func(ct *controllerTest) {
+				if err := utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("%v=true", scfeatures.OriginatingIdentity)); err != nil {
+					t.Fatalf("Failed to enable originating identity feature: %v", err)
+				}
+
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateSucceeded,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			verifyCondition: &v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionTrue,
+				Reason: "ProvisionedSuccessfully",
+			},
+			preCreateInstance: func(ct *controllerTest) {
+				catalogClient, err := changeUsernameForCatalogClient(ct.catalogClient, ct.catalogClientConfig, "instance-creator")
+				if err != nil {
+					t.Fatalf("could not change the username for the catalog client: %v", err)
+				}
+				ct.catalogClient = catalogClient
+				ct.client = catalogClient.ServicecatalogV1beta1()
+
+			},
+			postCreateInstance: func(ct *controllerTest) {
+				verifyUsernameInLastBrokerAction(ct.t, ct.osbClient, fakeosb.PollLastOperation, "instance-creator")
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ct := &controllerTest{
+				t:                            t,
+				broker:                       getTestBroker(),
+				instance:                     getTestInstance(),
+				skipVerifyingInstanceSuccess: tc.skipVerifyingInstanceSuccess,
```

Comment:
Might as well set `skipVerifyingInstanceSuccess` true for all test cases since the test is going to perform its own condition check in all cases anyway.

Reply:
The last test actually depends on it being false, but if you want me to get rid of that test I can. It's just that it's the only originating identity test present.

Reply:
The last one doesn't **need** to depend on that if you ran `verifyUsernameInLastBrokerAction` in the function supplied to `testController.run` instead, which I would argue is a marginally more appropriate place for it to be anyway.

It is fine the way you have it now, though.